pages/addboard.tsx

```diff
@@ -0,0 +1,92 @@
+import Button from '@/components/button/Button';
+import FileInput from '@/components/input/FileInput';
+import Input from '@/components/input/Input';
+import InputTextArea from '@/components/input/InputTextArea';
+import SectionTitle from '@/components/SectionTitle';
+import styles from '@/styles/AddBoard.module.scss';
+import { ChangeEvent, useEffect, useState } from 'react';
+
+const initialValues: PostArticleType = {
+  title: '',
+  content: '',
+};
+
+function AddBoardPage() {
+  const [isfilled, setIsFilled] = useState<boolean>(false);
+  const [values, setValues] = useState<PostArticleType>(initialValues);
+
+  const handleChange = (name: string, value: string) => {
+    setValues((prevValues) => ({
+      ...prevValues,
+      [name]: value,
+    }));
+  };
+
+  // 여기에서 <HTMLInputElement> 와 <HTMLTextAreaElement> 이 부분만 다른데 <T> 이런식으로 할 방법은 없을까요..?!!
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    handleChange(name, value);
+  };
+
+  const handleInputTextAreaChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    const { name, value } = e.target;
+    handleChange(name, value);
+  };
```

Comment:
한다면 
```suggestion
const handleInputChange = <T extends HTMLInputElement | HTMLTextAreaElement>(
  e: ChangeEvent<T>
) => {
  const { name, value } = e.target;
  handleChange(name, value);
};
```
이렇게도 가능할 것 같습니다!

혹은 event를 argument로 받지않고 name과 value를 받게하고 onChange에서 직접 익명함수로 호출할 수도 있겠구요!

Reply:
생각해보니 굳이 제네릭을 사용하지않고 `HTMLInputElement | HTMLTextAreaElement`로만 끝내도 여기서는 무방하구요!